doc/design/operator/operator_inheritance.md

```diff
@@ -0,0 +1,149 @@
+# Operator Design
+
+`Operator` in PaddlePaddle mainly describe how to do operation with `Variable`. It does not actually contains any data or state, but with key to find these `Variables/State` from `Scope`.
+
+Op will get/update data/state from Scope when running. It should not change the information inside op so all it's `Run()` function should be `const`
+
+Operator have a template parameter `DeviceContext`. DeviceContext is used to specify on which device this op will run. Each Op will implement multi Op according to different Context.
+
+#### We design the Operator class with tree layer.
+
+1. OperatorBase
+1. Operator
+1. CustomOperator
+
+
+#### `OperatorBase`
+
+`OperatorBase` cantains the base element of an Operator, but without any specialized information. The reason is that Net can treat all Ops as OperatorBase and donot need to consider the type or context of these ops. It can just call run() of Op:
```

Comment:
base element --> base class
cantains --> contains
specialized --> specific

The reason is that --> because
Net --> `NetworkBase`
donot --> do not
`run` --> `Run`